player/public/coreplugins/org.ekstep.epubrenderer-1.0/renderer/plugin.js

```diff
@@ -105,20 +100,35 @@ org.ekstep.contentrenderer.baseLauncher.extend({
         this.addEventHandlers();
         this.initProgressElements();
     },
+
+    // Get the total number of actual pages to render
+    // remove page from pagination if in <spine> <itemref> property is linear=no
+    getTotalPages: function() {
+        var instance = this
+        var data = instance.book.locations.spine
+        var array = []
+        for (index = 0; index < data.length; index++) {
+          if (data[index].hasOwnProperty('linear') && data[index].linear != "no") {
+            array[index] = data[index]
+          }
+        }
+        return array.length;
+    },
+
     addEventHandlers: function () {
         var instance = this;
         instance.book.generatePagination().then(function (data) {
             instance._start = data[0].cfi;
-            instance.totalPages = data.length;
+            instance.totalPages = instance.getTotalPages();
+            if(instance.totalPages <= 1) instance.lastPage = true; // if all pages are non linear or only one page is linear
```

Comment:
Why lessthan here. When instance.totalPages = 1 then instance.lastPage = true
When it is 0, we have to show message

Reply:
This is if spine has all the itemref as  **linear=no**, then our totalPages will be 0 and we always loads the first page then calculate the number of pages. This is corner case scenario. when total count page either 0 or 1